cai_causal_graph/graph_components.py

```diff
@@ -110,6 +111,44 @@ def variable_type(self, new_type: Union[NodeVariableType, str]):
             raise TypeError(f'Expected NodeVariableType or string, got object of type {type(new_type)}.')
         self._variable_type = new_type
 
+    def _check_var_name_is_valid(self, name: str):
+        """Check that the name is valid."""
+        vname, _ = get_variable_name_and_lag(name)
+        # do the same for the current node
+        vname_this, _ = get_variable_name_and_lag(self.identifier)
+
+        # check that the name is valid relative to the given Node
+        assert vname_this == vname, f'Invalid node name: {name}.'
+
+    @property
+    def variable_name(self) -> Optional[str]:
+        """Return the variable name of the node from the metadata."""
+        return self.meta.get('variable_name', None)
+
+    @variable_name.setter
+    def variable_name(self, new_name: str):
+        """
+        Set the variable name of the node.
+
+        :param new_name: New variable name.
+        """
+        self._check_var_name_is_valid(new_name)
+        self.meta['variable_name'] = new_name
+
+    @property
+    def time_lag(self) -> int:
```

Comment:
Haven't checked the full PR over but my instinct is that these shouldn't be here. We should have a TimeSeriesNode(Node) that inherits from the base node, that has these properties. Should only be used by TimeSeriesCausalGraph